packages/docmanager/src/dialogs.ts

```diff
@@ -196,3 +235,79 @@ namespace Private {
     return body;
   }
 }
+
+/**
+ * A widget used to name file on first save.
+ */
+class NameOnSaveHandler extends Widget {
+  /**
+   * Construct a new "name notebook file" dialog.
+   */
+  constructor(manager: IDocumentManager, oldPath: string) {
+    super({ node: Private.createNameFileNode(manager) });
+    this.addClass(FILE_DIALOG_CLASS);
+    const ext = PathExt.extname(oldPath);
+    const value = (this.inputNode.value = PathExt.basename(oldPath));
+    this.inputNode.setSelectionRange(0, value.length - ext.length);
+  }
+
+  /**
+   * Get the input text node.
+   */
+  get inputNode(): HTMLInputElement {
+    return this.node.getElementsByTagName('input')[0] as HTMLInputElement;
+  }
+
+  /**
+   * Get the value of the input widget.
+   */
+  getValue(): string {
+    return this.inputNode.value;
+  }
+
+  /**
+   * Get the checkbox node.
+   */
+  get checkboxNode(): HTMLInputElement {
+    return this.node.getElementsByTagName('input')[1] as HTMLInputElement;
+  }
+
+  /**
+   * Get checked of the checkbox widget.
+   */
+  getChecked(): boolean {
+    return this.checkboxNode.checked;
+  }
+}
+
+/**
+ * A namespace for private data.
+ */
+namespace Private {
+  /**
+   * Create the node for a rename after launch handler.
+   */
+  export function createNameFileNode(
+    manger: IDocumentManager,
+    translator?: ITranslator
+  ): HTMLElement {
+    translator = translator || nullTranslator;
+    const trans = translator.load('jupyterlab');
+    const body = document.createElement('div');
+    const name = document.createElement('input');
+    const checkbox = document.createElement('input');
+    const label = document.createElement('label');
+
+    checkbox.type = 'checkbox';
+    checkbox.addEventListener('change', function () {
+      manger.nameFileOnSave = !this.checked;
+    });
+    // Question: aglin two elements (flexbox?)
+    label.textContent = trans.__("Don't ask me again");
+    body.appendChild(name);
+    body.appendChild(checkbox);
+    body.appendChild(label);
```

Comment:
Should we add the label next to the checkbox @jess-x ?

Reply:
We should, and I was researching a way to do that, any suggestions @goanpeca ?

Reply:
Create a div and add to the div, and then add the div as a child?

Reply:
```ts
    const trans = translator.load('jupyterlab');
    const body = document.createElement('div');
    const name = document.createElement('input');
    const checkbox = document.createElement('input');
    const div = document.createElement('div');
    const label = document.createElement('label');

    checkbox.type = 'checkbox';
    checkbox.addEventListener('change', function () {
      manger.nameFileOnSave = !this.checked;
    });
    // Question: aglin two elements (flexbox?)
    label.textContent = trans.__("Don't ask me again");

    div.appendChild(checkbox);
    div.appendChild(label);

    body.appendChild(name);
    body.appendChild(div)
```